src/Microsoft.VisualStudio.Services.Agent/Blob/CustomerIntelligenceTelemetrySender.cs

```diff
@@ -50,19 +59,29 @@ public void SendActionTelemetry(ActionTelemetryRecord actionTelemetry)
                 var uploadStats = dedupRecord.UploadStatistics;
                 if (uploadStats != null)
                 {
-                    this._chunksUploaded += uploadStats.ChunksUploaded;
-                    this._compressionBytesSaved += uploadStats.CompressionBytesSaved;
-                    this._dedupUploadBytesSaved += uploadStats.DedupUploadBytesSaved;
-                    this._logicalContentBytesUploaded += uploadStats.LogicalContentBytesUploaded;
-                    this._physicalContentBytesUploaded += uploadStats.PhysicalContentBytesUploaded;
-                    this._totalNumberOfChunks += uploadStats.TotalNumberOfChunks;
+                    Interlocked.Add(ref this._chunksUploaded, uploadStats.ChunksUploaded);
```

Comment:
Why Interlocked here?  Seems like it may be expensive to do al these locks?

Reply:
Uploads are done in parallel. `+=` is not thread safe.

Reply:
I guess I'm wondering if we could take a lock around upload stats and avoid all the singular locks?

Reply:
Yeah we could do that. Might be a bit cleaner too. I'll make the change.